src/pages/controlled-form/controlled-form.tsx

```diff
@@ -0,0 +1,239 @@
+} from "../../redux/reducers/controlled-form-slice";
+import { useNavigate } from "react-router-dom";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { useEffect, useState } from "react";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { yupSchemaForHookForm } from "../../utils/yup-shema";
+import { setIsUncFormUpdate } from "../../redux/reducers/uncontrolled-form-slice";
+
+interface Form {
+  name: string;
+  age: string;
+  email: string;
+  pass1: string;
+  pass2: string;
+  country: string;
+  isMale: string;
+  isAgree: string;
+  image: Blob[] | {};
+}
+
+const ControlledForm = () => {
+  const dispatch = useDispatch();
+  const navigator = useNavigate();
+  const reader = new FileReader();
+  const formValues = useAppSelector((state) => state.controlledForm.formData);
+  const countries = useAppSelector((state) => state.countries.countries);
+  const [clueIsVisible, setClueIsVisible] = useState(false);
+  const [countryList, setCountryList] = useState<string[]>([]);
+  const [submitBtnClass, setSubmitBtnClass] = useState("btn_submit");
+
+  const { register, handleSubmit, getValues, setValue, formState } = useForm({
+    mode: "onChange",
+    defaultValues: {
+      name: formValues.name,
+      age: formValues.age,
+      email: formValues.email,
+      pass1: formValues.password,
+      pass2: formValues.password,
+      country: formValues.country,
+      isMale: formValues.isFemale ? "female" : formValues.isMale ? "male" : "",
+      isAgree: formValues.isAgree ? "yes" : "",
+      image: {},
+    },
+    resolver: yupResolver(yupSchemaForHookForm),
+  });
+
+  const { errors, isValid } = formState;
+
+  useEffect(() => {
+    if (isValid) {
+      setSubmitBtnClass("btn_submit");
+    } else setSubmitBtnClass("btn_submit_disabled");
+  }, [isValid]);
+
+  const submitDataToRedux = (data: Form) => {
+    dispatch(
+      setForm({
+        name: data.name,
+        age: data.age,
+        email: data.email,
+        password: data.pass1,
+        country: data.country,
+        isMale: data.isMale === "male" ? true : false,
+        isFemale: data.isMale === "female" ? true : false,
+        isAgree: data.isAgree === "yes" ? true : false,
+      }),
+    );
+    dispatch(setIsFilled(true));
+    dispatch(setIsUncFormUpdate(false));
+    dispatch(setIsContFormUpdate(true));
+    if (!Object.is(data.image, {})) {
+      const image = data.image as Blob[];
+      setDataBase(image[0]);
+    }
+    navigator("/");
+  };
+
+  const submitForm: SubmitHandler<Form> = async (data) => {
+    try {
+      await yupSchemaForHookForm.validate({
+        name: data.name,
+        age: data.age,
+        email: data.email,
+        pass1: data.pass1,
+        pass2: data.pass2,
+        country: data.country,
+        isAgree: data.isAgree,
+        image: data.image === null ? {} : data.image,
+      });
+      submitDataToRedux(data);
+    } catch (e) {
+      setSubmitBtnClass("btn_submit_disabled");
```

Comment:
This is just a tip, but in such cases it is better to separate the logic and put it in a separate error handling function
```suggestion
catch (e) {
handleFormSubmitError ()
} 
const handleFormSubmissionError = () => {
  setSubmitBtnClass("btn_submit_disabled");
};
     
```